File: javascript/crypto-square/crypto-square.js
export class Crypto {

    #text;

    constructor(inputText) {
        this.#text = inputText.toLowerCase().replaceAll(/\W/g, '');
    }

    get ciphertext() {

        if (!this.#text) {
            return '';
        }

        // finds the size of each word
        let sqrt = Math.sqrt(this.#text.length);
        let cols = Math.ceil(sqrt);

        // splits the text into 'cols-sized' chunks
        let regex = new RegExp(`.{1,${cols}}`, 'g');
        let arr = this.#text.match(regex).map(w => w.padEnd(cols, ' '));

        // maps the current text to the 'encrypted' one
        return [...Array(cols).keys()]
            .map(col => arr.map(word => word.charAt(col)).join('')).join(' ');
    }

}

File: python/crypto-square/crypto_square.py
import math
import re


def cipher_text(plain_text):

    # cleaning the input
    text = plain_text.lower()
    text = re.sub(r'[\W]', '', text)

    if not text:
        return text

    # size of each word
    cols = math.ceil(math.sqrt(len(text)))

    # separating into chunks and padding
    regex = ".{1," + str(cols) + "}"
    chunks = list(map(lambda w: w.ljust(cols), re.findall(regex, text)))

    # 'zipping' the chunks together and creating the return string
    return " ".join(["".join(w) for w in zip(*chunks)])
